### ISSUE TITLE
discussion：relink's filters don't seem to work in sub-filters（Questions about deleting entries for use of images）

### ISSUE THREAD

**Issue by Zacharia2** [flibbles/tw5-relink#43]
Hi there

I want to use relink's relink:backreferences[] filter to find images that have never been referenced. Then delete them to save storage space.

I did this: [all[tiddlers]!is[system]is[image]] :filter[relink:backreferences[]count[]compare:number:eq[0]] , filtering out all non-system image entries, giving sub-filters for each entry. Then filter out the image entries that have never been referenced.

But it doesn't work like the native backlinks[] filter.
Both writes have the same effect for the backlinks[] filter, but really don't have the same effect for relink:backreferences[].
[[Frodo]]:filter[relink:backreferences[]]
[[Frodo]relink:backreferences[]]

Only this writeup works; [[Frodo]relink:backreferences[]]

Have a nice life.

**Comment by flibbles** [comment]
Have you considered using the `orphans` operator?

`[relink:orphans[]!is[system]is[image]]`

Let me know if that works.

Farewell and Godspeed.

**Comment by flibbles** [comment]
I guess I should ask for clarifications. What do you mean by "writes"?

These two filters:
```
[[Frodo]]:filter[relink:backreferences[]]
```

```
[[Frodo]relink:backreferences[]]
```

...Should produce different results. The first will produce "Frodo" only if something references it. The second will produce all the tiddlers that reference Frodo. I've tested it. It works as expected for me.

**Comment by Zacharia2** [comment]
> 

Yes it is useful, thank you very much for your advice. What confused me was that different writing methods would lead to different effects, which I had not expected.

**Comment by Zacharia2** [comment]
> Have you considered using the `orphans` operator?
> 
> `[relink:orphans[]!is[system]is[image]]`
> 
> Let me know if that works.
> 
> Farewell and Godspeed.

Yes it is useful, thank you very much for your advice. What confused me was that different writing methods would lead to different effects, which I had not expected.

**Comment by Zacharia2** [comment]
> Have you considered using the `orphans` operator?
> 
> `[relink:orphans[]!is[system]is[image]]`
> 
> Let me know if that works.
> 
> Farewell and Godspeed.

Unfortunately, it didn't work. I created three new image entries. They are not referenced by other entries. When I enter this expression, the filter does not give results for these three entries.

**Comment by Zacharia2** [comment]
> I guess I should ask for clarifications. What do you mean by "writes"?
> 
> These two filters:
> 
> ```
> [[Frodo]]:filter[relink:backreferences[]]
> ```
> 
> ```
> [[Frodo]relink:backreferences[]]
> ```
> 
> ...Should produce different results. The first will produce "Frodo" only if something references it. The second will produce all the tiddlers that reference Frodo. I've tested it. It works as expected for me.

The meaning of writes is writing methods.
I use tiddlywiki's own filter to filter out entries with tags greater than two, which works fine. The filter expression: [all[tiddlers]!is[system]] :filter[tags[]count[]compare:number:gteq[2]].

If replaced with the expression: [all[tiddlers]!is[system]is[image]] :filter[backlinks[]count[]compare:number:eq[0]]
This also works fine, except that it only filters out the use of hard links "[[]]"

Following this idea, I associate perhaps relink:backreferences[] with the same effect. But the result didn't happen.

**Comment by Zacharia2** [comment]
So, what should I do to sift through all the entries, find the non-systematic image entries that are never used, and then delete them?

**Comment by flibbles** [comment]
I'm seeing a problem where images I import are not properly picked up by Relink. This is a bug I'll need to fix. However, it doesn't seem tied to images. When I save and reload, I'm able to pick those images up with `[relink:orphans[]]` then. Are you able to?

**Comment by Zacharia2** [comment]
> I'm seeing a problem where images I import are not properly picked up by Relink. This is a bug I'll need to fix. However, it doesn't seem tied to images. When I save and reload, I'm able to pick those images up with `[relink:orphans[]]` then. Are you able to?

[relink:orphans[]!is[system]is[image]]
This expression it works properly. The previous time I entered this expression it did not respond. This is really amazing, maybe it's a problem with my device.

**Comment by flibbles** [comment]
Not your device. Like I said, I am experiencing the problem too immediately after I import images. I'm glad it's working for you now, but there's a problem with Relink I'll need to resolve, so I'll leave this issue open until I can get around to it.

**Comment by flibbles** [comment]
This problem should be fully resolved as of Relink v2.3.0. You should no longer need to restart to have imported tiddlers and images show up in the orphan list. (except that $:/Imports might point to them, which makes them not orphans).

**Comment by Zacharia2** [comment]
thanks
